packages/drawer/src/ux-modal-service.ts

```diff
@@ -0,0 +1,277 @@
+    }
+  }
+
+  private setKeyListener() {
+    window.addEventListener('keyup', this);
+  }
+
+  private removeKeyListener() {
+    window.removeEventListener('keyup', this);
+  }
+
+  public handleEvent(event: KeyboardEvent) {
+    const key = this.getActionKey(event);
+    if (key === undefined) { return; }
+    const activeLayer = this.getLastDrawer();
+    const keyboard = activeLayer.keyboard;
+    if (key === 'Escape'
+      && (keyboard === true || keyboard === key || (Array.isArray(keyboard) && keyboard.indexOf(key) > -1))) {
+      activeLayer.dismiss();
+    } else if (key === 'Enter' && (keyboard === key || (Array.isArray(keyboard) && keyboard.indexOf(key) > -1))) {
+      activeLayer.ok();
+    }
+  }
+
+  private getActionKey(event: KeyboardEvent): DrawerKeybord | undefined {
+    if ((event.code || event.key) === 'Escape' || event.keyCode === 27) {
+      return 'Escape';
+    }
+    if ((event.code || event.key) === 'Enter' || event.keyCode === 13) {
+      return 'Enter';
+    }
+    return undefined;
+  }
+
+  public getLastLayer(): ModalLayer {
+    return this.modalLayers[this.modalLayers.length - 1];
+  }
+
+  public getLastDrawer(): UxDrawer {
+    return this.getLastLayer().drawer;
+  }
+
+  public get zIndex() {
+    return this.startingZIndex + this.modalLayers.length;
+  }
+
+  public open(options: ModalServiceOptions): UxDrawer & ModalServiceDrawer {
+    // const defaultConfig = this.container.get(DefaultDrawerConfiguration);
+    options = Object.assign({}, this.defaultConfig, options);
+    this.drawerIndex++;
+    const bindingContext: ModalBindingContext = {};
+    if (!options.viewModel && !options.view) {
+      throw new Error('Invalid Drawer Settings. You must provide "viewModel", "view" or both.');
+    }
+    const element = document.createElement('ux-drawer');
+    element.setAttribute('dismiss.delegate', 'dismiss()');
+    element.setAttribute('ok.delegate', 'ok($event)');
+    if (options.position !== undefined) {
+      element.setAttribute('position', options.position);
+    }
+    if (options.moveToBodyTag === false) {
+      element.setAttribute('move-to-body-tag.bind', 'false');
+    }
+    if (options.overlayDismiss === false) {
+      element.setAttribute('overlay-dismiss', 'false');
+    }
+    if (options.keyboard !== undefined) {
+      bindingContext.keyboard = options.keyboard;
+      element.setAttribute('keyboard.bind', 'keyboard');
+    }
+    if (typeof options.restoreFocus === 'function') {
+      bindingContext.restoreFocus = options.restoreFocus;
+      element.setAttribute('restore-focus.bind', 'restoreFocus');
+    }
+    
+    element.setAttribute('host.bind', 'false');
+    
+    if (options.theme) {
+      bindingContext.theme = options.theme;
+      element.setAttribute('theme.bind', `theme`);
+    }
+    const compose = document.createElement('compose');
+    compose.setAttribute('view-model.ref', 'composeViewModel');
+    element.innerHTML = compose.outerHTML;
```

Comment:
I think we can merge this line and the 2 above it into 1:
```ts
element.innerHTML = `<compose view-model.ref=composeViewModel>`;
```
it's shorter and easier to understand

Reply:
absolutely